Add macOS exporting and running pages. (#5105)
godotengine/godot-docs: tutorials/export/exporting_for_mac.rst

To notarize an app you mast have a valid `Apple Developer ID Certificate <https://developer.apple.com/>`__.

If you have an Apple Developer ID Certificate
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

- Enable "Code Signing", "Notarization", "Hardened Runtime" and "Timestamp" and disable "Debug" entitlement.
- Provided valid Apple ID credentials and certificate identity.

If "Notarization" is enable, Godot will automatically upload exported project for notarization.

You can use ``xcrun notarytool history`` command to check notarization status, and use ``xcrun notarytool log {ID}`` command to download notarization log.

If you encounter notarization issues, see `Resolving common notarization issues <https://developer.apple.com/documentation/security/notarizing_macos_software_before_distribution/resolving_common_notarization_issues>`__ for more info.

After notarization is completed, `staple the ticket <https://developer.apple.com/documentation/security/notarizing_macos_software_before_distribution/customizing_the_notarization_workflow>`__ to the exported project.

If you do not have an Apple Developer ID Certificate
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> Keep "Code Signing" enabled and leave "identinty" option empty, in this case Godot will use ad-hoc signature, which will make running an exported app easier for the end users.

Reviewer: ```suggestion
Keep ``Code Signing`` enabled and leave the ``Identity`` option empty, in this case Godot will use a ad-hoc signature, which will make running an exported app easier for the end users.
```

| Identity                     | The "Full Name" or "Common Name" of the signing identity, store in the macOS key chain. [1]_      |
+------------------------------+---------------------------------------------------------------------------------------------------+
| Timestamp                    | Requests a timestamp server to authenticate the time of signing. Required for notarization.       |
+------------------------------+---------------------------------------------------------------------------------------------------+
| Hardened Runtime             | Enables "Hardened Runtime". Required for notarization.                                            |
+------------------------------+---------------------------------------------------------------------------------------------------+
| Replace Existing Signature   | Replaces existing signatures of the GDNative libraries and embedded helper executables.           |
+------------------------------+---------------------------------------------------------------------------------------------------+

.. note::

>     To notarize an app, you must enable the "Hardened Runtime" and "Timestamp".

Reviewer: ```suggestion
    To notarize an app, you must enable the ``Hardened Runtime`` and ``Timestamp``.
```

> .. [1] Leave "identinty" option empty to use ad-hoc signature.

Reviewer: ```suggestion
.. [1] Leave ``Identity`` option empty to use ad-hoc signature.
```

+---------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Apple Events                          | [3]_ Enable to allow app to send Apple events to other apps.                                                                                                                                     |
+---------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Debugging                             | [4]_ You can temporarily enable this entitlement to use native debugger (GDB, LLDB) with the exported app. This entitlement should be disabled for production export.                            |
+---------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+

> .. [2] "Allow JIT Code Execution", "Allow Unsigned Executable Memory" and "Allow DYLD Environment Variables" entitlements are always enabled for the Godot Mono exports, and are not visible in the export options.

Reviewer: ```suggestion
.. [2] The ``Allow JIT Code Execution``, ``Allow Unsigned Executable Memory`` and ``Allow DYLD Environment Variables`` entitlements are always enabled for the Godot Mono exports, and are not visible in the export options.
```

> .. [3] These features aren't supported by Godot out of the box, enable it only if you are using add-ons which require it.

Reviewer: ```suggestion
.. [3] These features aren't supported by Godot out of the box, enable them only if you are using add-ons which require them.
```

> .. [4] To notarize an app, you must disable "Debugging" entitlement.

Reviewer: ```suggestion
.. [4] To notarize an app, you must disable the ``Debugging`` entitlement.
```

App Sandbox Entitlement
~~~~~~~~~~~~~~~~~~~~~~~

> App Sandbox restrict access to user's data, network and devices.

Reviewer: ```suggestion
The App Sandbox restricts access to user data, networking and devices.
```

Sandboxed apps can't access most of the file system, can't use custom file dialogs and execute binaries (using ``OS.execute`` and ``OS.create_process``) outside the ``.app`` bundle.
See `App Sandbox <https://developer.apple.com/documentation/security/app_sandbox?language=objc>`__ for more info.

.. note::

>     To distribute an app through the App Store, you must enable App Sandbox.

Reviewer: ```suggestion
    To distribute an app through the App Store, you must enable the App Sandbox.
```
